core/src/main/java/com/nukesz/game/screen/GameScreen.java

```diff
@@ -49,25 +51,27 @@
     private GlyphLayout layout = new GlyphLayout();
     private Texture snakeHead;
     private Texture snakeBody;
-    private Texture apple;
     private Texture wall;
-    private boolean appleAvailable = false;
-    private int appleX;
-    private int appleY;
+    Apple appleObj;
     private Array<BodyPart> bodyParts = new Array<>();
     private int snakeXBeforeUpdate;
     private int snakeYBeforeUpdate;
     private boolean directionSet = false;
     private State state = State.INITIAL;
     private int score = 0;
     private Array<Wall> walls;
-
+    OnscreenControlRenderer controlRenderer;
+    private boolean isSideControllerNeeded =true;
     public GameScreen(SnakeGame game) {
         super(game);
+        if (isSideControllerNeeded) {
```

Comment:
We could show always the controller. No need for a boolean variable.

Reply:
The boolean value to be a place holder if we only want to show it just for Android.